lib/shopify_api/rest.ex

```diff
@@ -11,25 +11,74 @@ defmodule ShopifyAPI.REST do
   alias ShopifyAPI.JSONSerializer
   alias ShopifyAPI.REST.Request
 
-  @doc false
-  def get(%AuthToken{} = auth, path, params \\ []) do
-    Request.perform(auth, :get, path, "", params)
+  @doc """
+  Underlying utility retrieval function. The options passed affect both the
+  return value and, ultimately, the number of requests made to Shopify.
+  Options:
+    * `:pagination` - Can be `:none`, `:stream`, or `:auto`. Defaults to :auto
+      `:none` will only return the first page. You won't have access to the headers to manually
+      paginate.
+      `:auto` will block until all the pages have been retrieved and concatenated together.
+      `:stream` will return a `Stream`, prepopulated with the first page.
+  """
+  @spec get(AuthToken.t(), path :: String.t(), keyword(), keyword()) ::
+          {:ok, %{required(String.t()) => [map()]}} | Enumerable.t()
+  def get(%AuthToken{} = auth, path, params \\ [], options \\ []) do
+    collect_results = fn
+      %{} = result, {:ok, acc} -> {:cont, {:ok, [result | acc]}}
+      error, {:ok, _acc} -> {:halt, error}
+    end
+
+    case pagination(options) do
+      :none ->
+        with {:ok, response} <- Request.perform(auth, :get, path, "", params) do
+          {:ok, fetch_body(response)}
+        end
+
+      :stream ->
+        Request.stream(auth, path, params)
+
+      _auto_or_nil ->
```

Comment:
`pagination/1` can't return a nil value, AFAICT - should this just be `:auto` instead?